sdk/containerregistry/Azure.Containers.ContainerRegistry/src/Blob/ContainerRegistryBlobClient.cs

```diff
@@ -0,0 +1,503 @@
+        }
+
+        /// <summary> Initializes a new instance of ContainerRegistryBlobClient for mocking. </summary>
+        protected ContainerRegistryBlobClient()
+        {
+        }
+
+        /// <summary>
+        /// Gets the registry service endpoint for this client.
+        /// </summary>
+        public virtual Uri Endpoint => _endpoint;
+
+        /// <summary>
+        /// Gets the name of the repository that logically groups the artifact parts.
+        /// </summary>
+        public virtual string RepositoryName => _repositoryName;
+
+        /// <summary>
+        /// Upload an artifact manifest from a stream.
+        /// </summary>
+        /// <param name="stream">The stream containing the serialized manifest.</param>
+        /// <param name="options">Options for configuring the upload operation.</param>
+        /// <param name="cancellationToken"> The cancellation token to use. </param>
+        /// <returns></returns>
+        public virtual Response<UploadManifestResult> UploadManifest(Stream stream, UploadManifestOptions options = default, CancellationToken cancellationToken = default)
+        {
+            Argument.AssertNotNull(stream, "stream");
+
+            options ??= new UploadManifestOptions();
+
+            var manifest = DeserializeManifest(stream);
+            return UploadManifest(manifest, options, cancellationToken);
+        }
+
+        /// <summary>
+        /// Upload an artifact manifest from a stream.
+        /// </summary>
+        /// <param name="stream">The stream containing the serialized manifest.</param>
+        /// <param name="options">Options for configuring the upload operation.</param>
+        /// <param name="cancellationToken"> The cancellation token to use. </param>
+        /// <returns></returns>
+        public async virtual Task<Response<UploadManifestResult>> UploadManifestAsync(Stream stream, UploadManifestOptions options = default, CancellationToken cancellationToken = default)
+        {
+            Argument.AssertNotNull(stream, "stream");
+
+            options ??= new UploadManifestOptions();
+
+            var manifest = DeserializeManifest(stream);
+            return await UploadManifestAsync(manifest, options, cancellationToken).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Upload a manifest for an OCI Artifact.
+        /// </summary>
+        /// <param name="manifest">The manifest to upload.</param>
+        /// <param name="options">Options for configuring the upload operation.</param>
+        /// <param name="cancellationToken"> The cancellation token to use. </param>
+        /// <returns></returns>
+        public virtual Response<UploadManifestResult> UploadManifest(OciManifest manifest, UploadManifestOptions options = default, CancellationToken cancellationToken = default)
+        {
+            Argument.AssertNotNull(manifest, "manifest");
+
+            options ??= new UploadManifestOptions();
```

Comment:
nit - if this could be a hot path, perhaps we could re-use a default instance and create it just once?

Reply:
I'll file an issue.

Reply:
https://github.com/Azure/azure-sdk-for-net/issues/23833